frontend/src/tests/ForestClientInput.spec.ts

```diff
@@ -75,31 +107,297 @@ describe('ForestClientInput', () => {
+        expect(inputField.value).toBe('');
+    });
+
+    it('should add active forest client numbers and raise errors for inactive and invalid numbers in multiple input', async () => {
+        // Mutiple inputs: two valid ones, one invalid and one inactive
+        const mixedInputs = `${TEST_SUCCESS_FOREST_CLIENT_NUMBER},${TEST_INACTIVE_FOREST_CLIENT_NUMBER},${TEST_INVALID_FOREST_CLIENT_NUMBER},${TEST_SUCCESS_FOREST_CLIENT_NUMBER_2}`
+        await input.setValue(mixedInputs);
+        expect(inputField.value).toBe(mixedInputs);
+
+        await wrapper.find("[aria-label='Add Client Numbers']").trigger('click');
+        await flushPromises();
+
+        const setVerifiedForestClients = wrapper.emitted(
+            'setVerifiedForestClients'
+        );
+        expect(wrapper.emitted()).toHaveProperty('setVerifiedForestClients');
+        // has been called twice because we have two active ones
+        expect(setVerifiedForestClients).toHaveLength(2);
+
+        const validationError = wrapper.findAll('#forestClientInputValidationError');
+        // we need to have two error messages as well (inactive and invalid)
+        expect(validationError).toHaveLength(2);
+        // The first entry in the forest client card should be equal to the first active input
+        expect(setVerifiedForestClients![0][0]).toEqual(TEST_SUCCESS_FOREST_CLIENT_NUMBER);
+        // The second entry in the forest client card should be equal to the second active input
+        expect(setVerifiedForestClients![1][0]).toEqual(TEST_SUCCESS_FOREST_CLIENT_NUMBER_2);
+        // the forest client information card shows and verify the information
+        expect(wrapper.html().includes('Verified Client ID information')).toBe(
+            true
+        );
+        expect(
+            wrapper
+                .find('#forest-client-id')
+                .text()
+                .includes(TEST_SUCCESS_FOREST_CLIENT_NUMBER)
+        ).toBe(true);
+        expect(
+            wrapper
+                .find('#forest-client-name')
+                .text()
+                .includes(TEST_FOREST_CLIENT_NAME)
+        ).toBe(true);
+        expect(
+            wrapper.find('#forest-client-status').text().includes('Active')
+        ).toBe(true);
+        // the forest client input should still display the error input numbers
+        expect(inputField.value).toContain(TEST_INACTIVE_FOREST_CLIENT_NUMBER);
+        expect(inputField.value).toContain(TEST_INVALID_FOREST_CLIENT_NUMBER);
+
+        const clientIdCardList = wrapper.findAll('#forest-client-id');
+
+        expect(clientIdCardList).toHaveLength(2);
+
+        expect(clientIdCardList[0].text().includes(TEST_SUCCESS_FOREST_CLIENT_NUMBER)).toBe(true);
+        expect(clientIdCardList[1].text().includes(TEST_SUCCESS_FOREST_CLIENT_NUMBER_2)).toBe(true);
+    });
+
+    //TODO - The test case below needs to adjust the mockImplementation for the api call to have a non blocking delay
```

Comment:
Hi @NickSaglioni, if this case is very tricky and had already spent lots of time, should we do this in separate ticket to add and remove this commented out section? Thanks.
@MCatherine1994

Reply:
so here, we can mock the api a different way and mount another wrapper just for this test case,

```
 it('should test the adding button in on loading while we call the api', async () => {
        vi.spyOn(
            AppActlApiService.forestClientsApi,
            'search'
        ).mockImplementation(async (item) => {
            // here we suppose while we call backend api, the loading state will be true
            setLoadingState(true);
            return forestClientsApiSearchMock(item);
        });
        const wrapper = mount(ForestClientInput, {
            props: {
                userId: 'testUser',
            },
        });
        const input = wrapper.find('#forestClientInput');
        const inputField = input.element as HTMLInputElement;
        await input.setValue(TEST_SUCCESS_FOREST_CLIENT_NUMBER);
        expect(inputField.value).toBe(TEST_SUCCESS_FOREST_CLIENT_NUMBER);
        await wrapper
            .find("[aria-label='Add Client Numbers']")
            .trigger('click');
        expect(isLoading()).toBe(true);
        expect(
            wrapper.find("[aria-label='Add Client Numbers']").element
                .textContent
        ).contains('Loading');

        // cleanup
        setLoadingState(false);
        vi.clearAllMocks();
        wrapper.unmount();
}
```